spec/app/ics-070-shared-queue/README.md

```diff
@@ -0,0 +1,485 @@
+# Queue Module
+
+## Introduction
+
+This document presents a technical specification of the **queue module**.
+The queue module contains a logic necessary for implementing shared FIFO queue among two parties (i.e., two blockchains).
+Therefore, in an implementation of the FIFO queue we are interested in, the following entities take place:
+  - Two blockchains: Each validator of each blockchain implements its queue module.
+  We say that one blockchain is the parent blockchain, whereas the other is the baby blockchain.
+  - IBC communication: There exists an IBC communication among two aforementioned blockchains.
+  Blockchains communicate exclusively using the IBC channels among them.
+
+We present two versions of the queue module.
+The versions provide slightly different interface and guarantees, as we describe in the following subsection.
+
+### Shared FIFO Queue Specification
+
+Our shared FIFO queue is a concurrent object.
+We now define both versions of our queue.
+
+#### Sequential Consistent FIFO Queue
```

Comment:
Maybe I am being dense, but this seems a long description of exisiting ordered channels